pkg/replication/order_test.go

```diff
@@ -0,0 +1,45 @@
+package replication
+
+import (
+	"testing"
+	"time"
+
+	. "github.com/anthonybishopric/gotcha"
+)
+
+func TestMaxOrderBlocks(t *testing.T) {
+	order, err := NewMaxOrder([]string{"a", "b", "c", "d", "e"}, 2, nil)
+	Assert(t).IsNil(err, "should have been nil error")
+	next := make(chan string)
+	done := make(chan string)
+	errs := make(chan error)
+	quit := make(chan struct{})
+
+	go order.Execute(next, done, errs, quit)
+
+	count := 0
+OUTER:
+	for {
+		select {
+		case <-next:
+			count++
+		case <-time.After(500 * time.Millisecond):
```

Comment:
I don't know of one. Select ordering is pseudorandom, so the solution you suggested earlier would still be random (as is this test, though, so it's not like this is much better). The only way to really know that the channel has "stopped" is to block on it until you feel confident that it's going to keep blocking in the foreseeable future.

fwiw, the real purpose of the test is to make sure that the `active` constraint doesn't get violated. This particular test verifies the constraint by waiting for the `next` channel to "block". I don't think there's any really deterministic way to test this, but if there's a less time-sensitive option than this one, I'm for it.